Merge release/18.1 into develop (code freeze)

AliSoftware opened the pull request:
18.1 code freeze just done for WP + JP.

AliSoftware committed:
Bump version number

AliSoftware committed:
Bump Jetpack version number

AliSoftware committed:
Update draft release notes for Jetpack 18.1.

AliSoftware committed:
Release Notes: add new section for next version (18.2)

AliSoftware committed:
Update draft release notes for 18.1.

AliSoftware committed:
Update strings for translation

AliSoftware committed:
Send strings to translation.

AliSoftware committed:
Bump FluxC to 1.24.0

loremattei commented on version.properties:
```diff
@@ -1,12 +1,12 @@
 # Jetpack version information
-jetpack.versionName=18.0
-jetpack.versionCode=1096
+jetpack.versionName=18.1-rc-1
```
I'm not familiar with the Jetpack setup, so: why the old version code is different from the one for WordPress (`1096` vs `1098`)?

AliSoftware replied:
Because paaHJt-2s8-p2 is still pending 😉 

That's the whole issue with how we've been managing those versions automatically so far after Javon's changes in the 1.3.0 toolkit: versionName / versionCode for WP and JP are still handled and bumped separately by the toolkit.  
And since WP has both an alpha and a beta, bumping versionCodes 2-by-2, while JP only has beta and bumps its versionCodes 1-by-1, that creates that drift.

I sometimes realign the versionCodes manually from time to time, but it's still not great to not have that automated and them using the same code for both… hence why I'm eager to get this paaHJt-2s8-p2 project (and its followups) done, to stop having to handle that all manually every beta/release. (Draft for fixing this [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15230))

AliSoftware replied:
Just to give you an idea of how it went with last releases because of the current state of automation and versionCodes being handled separately by the toolkit (the thing that the project plans to fix):

| versionName | WP prod/beta | WP alpha | JP prod/beta | Comment |
|:--|--:|--:|--:|:--|
| 17.9 | 1090 | - | 1086 | Existing drift from previous betas |
| 18.0-rc-1 | 1091 | 1092 | 1091 | JP/WP versionCodes manually realigned on code freeze |
| 17.9.1 | 1093 | - | 1093 | Manually fixed JP versionCode for the hotfix to be 1093 like WP and not just 1091+1 |
| 18.0-rc-2 | 1094 | 1095 | 1094 |
| 18.0-rc-3 | 1096 | 1097 | 1095 | JP drift as it has no alpha and is bumped independently from WP |
| 18.0 | 1098 | - | 1096 | |
| 18.1-rc-1 | 1099 | 1100 | 1099 | Realigned manually on code freeze (this PR) |

loremattei reviewed (approved):
This looks good to me. 

I left a small question which should be an issue.

The Peril error seems to be related to your recent changes to the configuration. I this that the label check didn't get triggered before on Release PRs. If we can't manage to skip it with the new configuration, we can update the rule to skip the `Label Type` check when the `Releases` label is applied.

AliSoftware commented:
> The Peril error seems to be related to your recent changes to the configuration. I this that the label check didn't get triggered before on Release PRs. If we can't manage to skip it with the new configuration, we can update the rule to skip the Label Type check when the Releases label is applied.

This was indeed due to the recent changes from config, but because we accidentally made the PR events trigger `issue/label.ts` instead of `pr/label.ts` in WPAndroid, so it was running the incorrect rule. I just fixed this in https://github.com/Automattic/peril-settings/pull/86. Now that the fix is merged, I added then removed the `Tooling` label here to re-trigger Peril and everything passed, confirming the config is fixed.
